openstf/model/objective.py

```diff
@@ -63,9 +63,9 @@ def __call__(self, trial: optuna.trial.FrozenTrial) -> float:
             back_test=True,
         )
         # Split in x, y data (x are the features, y is the load)
-        train_x, train_y = train_data.iloc[:, 1:], train_data.iloc[:, 0]
-        valid_x, valid_y = validation_data.iloc[:, 1:], validation_data.iloc[:, 0]
-        test_x, test_y = test_data.iloc[:, 1:], test_data.iloc[:, 0]
+        train_x, train_y = train_data.iloc[:, 1:-1], train_data.iloc[:, 0]
+        valid_x, valid_y = validation_data.iloc[:, 1:-1], validation_data.iloc[:, 0]
+        test_x, test_y = test_data.iloc[:, 1:-1], test_data.iloc[:, 0]
```

Comment:
Since order of columns is very important here, perhaps it's a good idea to add a check and make sure that the last columns is 'Horizon' and should be omitted

Reply:
I agree, this looks like black magic to me. We could probably think about this a bit more to have a better understanding on how and when the structure of our data changes and or is required in a different form. But to keep it simple for now we should (like Frank said) probably check if the column is there and remove it and or set the slice depending on the outcome. (assuming both are valid use cases; with and without a Horizons column)

Reply:
Check added